libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranularPermissions/GranularGranteeGroupItem.tsx

```diff
@@ -1,31 +1,51 @@
 // (C) 2022-2023 GoodData Corporation
 
-import React, { useMemo } from "react";
+import React, { useCallback, useMemo } from "react";
 import { useIntl } from "react-intl";
 import cx from "classnames";
 
 import { GranteeGroupIcon } from "../GranteeIcons";
-import { GranteeItem, IGranularGranteeGroup } from "../types";
+import { DialogModeType, GranteeItem, IGranularGranteeGroup } from "../types";
 import { getGranteeItemTestId, getGranteeLabel } from "../utils";
 import { CurrentUserPermissions } from "../../types";
 
 import { GranularPermissionsDropdownWithBubble } from "./GranularPermissionsDropdown";
 import { usePermissionsDropdownState } from "./usePermissionsDropdownState";
 import { getGranteePossibilities } from "./permissionsLogic";
+import { useComponentInteractionContext } from "../ComponentInteractionContext";
 
 interface IGranularGranteeGroupItemProps {
     grantee: IGranularGranteeGroup;
     currentUserPermissions: CurrentUserPermissions;
     isDashboardLocked: boolean;
+    mode: DialogModeType;
     onChange: (grantee: GranteeItem) => void;
     onDelete: (grantee: GranteeItem) => void;
 }
 
 export const GranularGranteeGroupItem: React.FC<IGranularGranteeGroupItemProps> = (props) => {
-    const { grantee, currentUserPermissions, isDashboardLocked, onChange, onDelete } = props;
+    const { grantee, currentUserPermissions, isDashboardLocked, onChange, onDelete, mode } = props;
+    const { permissions } = grantee;
     const { isDropdownOpen, toggleDropdown } = usePermissionsDropdownState();
+    const onInteraction = useComponentInteractionContext();
     const intl = useIntl();
 
+    const handleToggleDropdown = useCallback(() => {
+        toggleDropdown();
```

Comment:
Could you extract this logic from component ? Separate hook? or util function?. And move this closer to permissions ?

Main purpose of this callback is hidden by  onInteraction handling

Reply:
Maybe ComponentInteractionContext.tsx could be right place

Reply:
done